components/tools/OmeroJava/test/integration/chown/PermissionsTest.java

```diff
@@ -467,6 +471,571 @@ public void testChownAnnotationPrivate(boolean isDataOwner, boolean isAdmin, boo
+            }
+        }
+
+        /* Another user (otherImporter) links the first user/importer's 
+         * annotationsOwnForTripleLinking to a second image (otherImage) 
+         * which belongs to this other user (otherImporter).
+         * as well as to the first image (image) which belongs to the importerTargetUser
+         * Note that ROI and thumbnail cannot be multiply linked, and thus will
+         * remain just singly linked objects in the annotationsOwnForTripleLinking batch
+         * of annotations
+         * Again, do the same sequence of linking for both sets of annotations/images, 1 and 2.*/
+        init(otherImporter1);
+        /* check group permissions */
+        if (users1CanAnnotateOthers) {
+            for (final IObject annotation : annotationsOwnForTripleLinking1) {
+                if (!(annotation instanceof Roi) & !(annotation instanceof Thumbnail) & !(annotation instanceof RectangleI)) {
+                    final ImageAnnotationLink linkOtherImage = (ImageAnnotationLink) annotateImage(otherImage1, (Annotation) annotation);
+                    linksOthersToOwnAnnOthersImage1.add((ImageAnnotationLink) linkOtherImage.proxy());
+                    final ImageAnnotationLink linkOwnImage = (ImageAnnotationLink) annotateImage(image1, (Annotation) annotation);
+                    linksOthersToOwnAnnOwnImage1.add((ImageAnnotationLink) linkOwnImage.proxy());
+                }
+            }
+        }
+        init(otherImporter2);
+        if (users2CanAnnotateOthers) {
+            for (final IObject annotation : annotationsOwnForTripleLinking2) {
+                if (!(annotation instanceof Roi) & !(annotation instanceof Thumbnail) & !(annotation instanceof RectangleI)) {
+                    final ImageAnnotationLink linkOtherImage = (ImageAnnotationLink) annotateImage(otherImage2, (Annotation) annotation);
+                    linksOthersToOwnAnnOthersImage2.add((ImageAnnotationLink) linkOtherImage.proxy());
+                    final ImageAnnotationLink linkOwnImage = (ImageAnnotationLink) annotateImage(image2, (Annotation) annotation);
+                    linksOthersToOwnAnnOwnImage2.add((ImageAnnotationLink) linkOwnImage.proxy());
+                }
+            }
+        }
+
+        /* create two tag sets, one as the first user
+         * (importerTargetUser) and another one as the other user
+         * and create two tags as the first user and one tag
+         * as the other user */
+        init(importerTargetUser1);
+        final List<TagAnnotation> tagsetOwn = createTagsets(1);
+        init(otherImporter1);
+        final List<TagAnnotation> tagsetOthers = createTagsets(1);
+        init(importerTargetUser1);
+        final List<TagAnnotation> tagsOwn = createTags(2);
+        init(otherImporter1);
+        final List<TagAnnotation> tagsOthers = createTags(1);
+
+        /* group the tagsets to one pot in order to be able to reuse the 
+         * definteLinikingTags method
```

Comment:
Discussed with @pwalczysko: "tagset" section might be removed. This was in the initial test which became quite complex